diff --git a/cpp/src/IceGrid/Database.cpp b/cpp/src/IceGrid/Database.cpp
index 5e5ff94..b2d91a0 100644
--- a/cpp/src/IceGrid/Database.cpp
+++ b/cpp/src/IceGrid/Database.cpp
@@ -395,7 +395,7 @@ Database::syncApplicationDescriptor(const ApplicationDescriptor& newDesc, AdminS
         Lock sync(*this);
         checkSessionLock(session);
 
-        while(_updating.find(update.descriptor.name) != _updating.end())
+        while(_updating.find(newDesc.name) != _updating.end())
         {
             wait();
         }
